app/server.rb

```diff
@@ -0,0 +1,62 @@
+require 'sinatra/base'
+require 'rack-flash'
+require_relative 'data_mapper_setup'
+require_relative './models/user'
+require_relative './session_helper'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  set :session_secret, 'superbly secret'
+  use Rack::Flash
+  use Rack::MethodOverride
+  include SessionHelper
+
+  get '/' do
+    erb :index
+  end
+
+  post '/users' do
+    user = User.new(email: params[:email],
+                    name: params[:name],
+                    password: params[:password],
+                    username: params[:username])
```

Comment:
We want our controllers to be as thin as possible ->

```ruby
user = User.create_from_params!(params)
```

Then all the `initializing` will be in the `User` class, not in the controller here.